packages/sdk/contractkit/src/wrappers/GrandaMento.ts

```diff
@@ -0,0 +1,143 @@
+}
+
+export interface ExchangeProposal {
+  exchanger: string
+  stableToken: string
+  sellAmount: BigNumber
+  buyAmount: BigNumber
+  approvalTimestamp: BigNumber
+  state: ExchangeProposalState
+  sellCelo: boolean
+}
+
+type AllStableConfig = Map<StableTokenContract, StableTokenExchangeLimits>
+
+export class GrandaMentoWrapper extends BaseWrapper<GrandaMento> {
+  approver = proxyCall(this.contract.methods.approver)
+
+  spread = proxyCall(this.contract.methods.spread, undefined, fixidityValueToBigNumber)
+
+  vetoPeriodSeconds = proxyCall(
+    this.contract.methods.vetoPeriodSeconds,
+    undefined,
+    valueToBigNumber
+  )
+
+  owner = proxyCall(this.contract.methods.owner)
+
+  getActiveProposalIds = proxyCall(this.contract.methods.getActiveProposalIds)
+
+  setStableTokenExchangeLimits = proxySend(
+    this.kit,
+    this.contract.methods.setStableTokenExchangeLimits
+  )
+
+  approveExchangeProposal = proxySend(this.kit, this.contract.methods.approveExchangeProposal)
+
+  executeExchangeProposal = proxySend(this.kit, this.contract.methods.executeExchangeProposal)
+  cancelExchangeProposal = proxySend(this.kit, this.contract.methods.cancelExchangeProposal)
+
+  async createExchangeProposal(
+    stableTokenRegistryId: StableTokenContract,
+    sellAmount: BigNumber,
+    sellCelo: true
+  ) {
+    const createExchangeProposalInner = proxySend(
+      this.kit,
+      this.contract.methods.createExchangeProposal
+    )
+    return createExchangeProposalInner(stableTokenRegistryId, sellAmount.toNumber(), sellCelo)
+  }
+
+  async getExchangeProposal(exchangeProposalID: string | number): Promise<ExchangeProposal> {
+    const result = await this.contract.methods.exchangeProposals(exchangeProposalID).call()
+    const state = parseInt(result.state, 10)
+
+    if (state === ExchangeProposalState.None) {
+      throw new Error("Proposal doesn't exist")
+    }
+
+    return {
+      exchanger: result.exchanger,
+      stableToken: result.stableToken,
+      sellAmount: new BigNumber(result.sellAmount),
+      buyAmount: new BigNumber(result.buyAmount),
+      approvalTimestamp: new BigNumber(result.approvalTimestamp),
+      sellCelo: result.sellCelo,
+      state,
+    }
+  }
+
+  async stableTokenExchangeLimits(
+    stableTokenTymbol: StableToken
+  ): Promise<StableTokenExchangeLimits> {
+    const stableTokenRegistryId = this.kit.celoTokens.getContract(stableTokenTymbol)
+    const result = await this.contract.methods
+      .stableTokenExchangeLimits(stableTokenRegistryId.toString())
+      .call()
+    return {
+      minExchangeAmount: new BigNumber(result.minExchangeAmount),
+      maxExchangeAmount: new BigNumber(result.maxExchangeAmount),
+    }
+  }
+
+  async getAllStableTokenLimits(): Promise<AllStableConfig> {
+    const out: AllStableConfig = new Map()
+
+    const res = await Promise.all(
+      Object.values(StableToken).map((key) => this.stableTokenExchangeLimits(key))
+    )
+    await Promise.all(
```

Comment:
This looks wrong to me.

`this.kit.celoTokens.getContract` returns a promise. So you're really doing `out.set(Promise, res[index]`.

I think

```
    await Promise.all(
      Object.values(StableToken).map(async (key, index) =>
        out.set(await this.kit.celoTokens.getContract(key), res[index])
      )
    )
```

is what you want.

Reply:
It doesn't seem to return a promise: https://github.com/celo-org/celo-monorepo/blob/master/packages/sdk/contractkit/src/celo-tokens.ts#L233

Reply:
My mistake, you are correct. I was looking at the ContractCache `getContract` method.

In that case we can just get rid of the `Promise.all`

Reply:
you're right, surprised it compiles anyway!